src/GitHub.App/Services/GitClient.cs

```diff
@@ -424,16 +424,7 @@ public Task<bool> IsHeadPushed(IRepository repo)
 
             return Task.Factory.StartNew(() =>
             {
-                if (repo.Head.IsTracking)
-                {
-                    var trackedBranchTip = repo.Head.TrackedBranch.Tip;
-                    if (trackedBranchTip != null)
-                    {
-                        return repo.Head.Tip.Sha == trackedBranchTip.Sha;
-                    }
-                }
-
-                return false;
+                return repo.Head.TrackingDetails.AheadBy == 0;
```

Comment:
Ah, I didn't realize `AheadBy` was an `int?`!

Maybe do an explicit `if (repo.Head.TrackingDetails.AheadBy == null) return false;` for clarity?